examples/energy-management-app/energy-management-common/src/EnergyEvseTargetsStore.cpp

```diff
@@ -0,0 +1,478 @@
+
+    return CHIP_NO_ERROR;
+}
+
+const DataModel::List<const Structs::ChargingTargetScheduleStruct::Type> & EvseTargetsDelegate::GetTargets()
+{
+    return mChargingTargetSchedulesList;
+}
+
+/* static */
+uint16_t EvseTargetsDelegate::GetTlvSizeEstimate()
+{
+    size_t kListOverhead = 4;
+    size_t chargingTargetStuctEstimate =
+        TLV::EstimateStructOverhead(sizeof(uint16_t), sizeof(Optional<chip::Percent>), sizeof(Optional<int64_t>));
+    size_t chargingTargetScheduleStructEstimate = TLV::EstimateStructOverhead(sizeof(chip::BitMask<TargetDayOfWeekBitmap>)) +
+        kListOverhead + kEvseTargetsMaxTargetsPerDay * chargingTargetStuctEstimate;
+    size_t totalEstimate = kEvseTargetsMaxNumberOfDays * chargingTargetScheduleStructEstimate + kListOverhead;
+
+    return static_cast<uint16_t>(totalEstimate);
+}
+
+CHIP_ERROR EvseTargetsDelegate::LoadTargets()
+{
+    // The DataModel::List<const Structs::ChargingTargetScheduleStruct::Type> data structure contains a list of
+    // ChargingTargetScheduleStructs which in turn contains a list of ChargingTargetStructs. Lists contain pointers
+    // to objects allocated outside of the List. For mChargingTargetSchedulesList, that memory is allocated in
+    // mChargingTargets and mChargingTargetSchedulesArray.
+
+    Platform::ScopedMemoryBuffer<uint8_t> backingBuffer;
+    uint16_t length = GetTlvSizeEstimate();
+    ReturnErrorCodeIf(!backingBuffer.Calloc(length), CHIP_ERROR_NO_MEMORY);
+
+    CHIP_ERROR err = mpTargetStore->SyncGetKeyValue(spEvseTargetsKeyName, backingBuffer.Get(), length);
+    if (err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
+    {
+        // Targets does not exist persistent storage -> initialise mChargingTargetSchedulesList as empty
+        mChargingTargetSchedulesList = DataModel::List<const Structs::ChargingTargetScheduleStruct::Type>();
+
+        return CHIP_NO_ERROR;
+    }
+    ReturnErrorOnFailure(err);
+
+    TLV::ScopedBufferTLVReader reader(std::move(backingBuffer), length);
+
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::AnonymousTag()));
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(reader.EnterContainer(arrayType));
+
+    uint16_t chargingTargetSchedulesIdx = 0;
+    while ((err = reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag())) == CHIP_NO_ERROR)
+    {
+        TLV::TLVType evseTargetEntryType;
+
+        ReturnErrorOnFailure(reader.EnterContainer(evseTargetEntryType));
+
+        // DayOfWeek bitmap
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TargetEntryTag::kDayOfWeek)));
+        ReturnErrorOnFailure(reader.Get(mChargingTargetSchedulesArray[chargingTargetSchedulesIdx].dayOfWeekForSequence));
+
+        ChipLogProgress(AppServer, "LoadTargets: DayOfWeekForSequence = 0x%02x",
+                        mChargingTargetSchedulesArray[chargingTargetSchedulesIdx].dayOfWeekForSequence.GetField(
+                            static_cast<TargetDayOfWeekBitmap>(0x7F)));
```

Comment:
we should have a constant like `kAllTargetDaysMask` to use this and everywhere else for the GetField mask.

Reply:
kAllTargetDaysMask is now defined in examples/energy-management-app/energy-management-common/include/EnergyEvseDelegateImpl.h and used appropriately.